File: app/src/main/java/com/example/loren/minesample/fragment/HomeFragment.java
package com.example.loren.minesample.fragment;

import android.app.ActivityOptions;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.example.loren.minesample.ChatActivity;
import com.example.loren.minesample.CopyWxPullActivity;
import com.example.loren.minesample.CustomViewActivity;
import com.example.loren.minesample.FlagActivity;
import com.example.loren.minesample.MoveViewActivity;
import com.example.loren.minesample.OpenScreenActivity;
import com.example.loren.minesample.R;
import com.example.loren.minesample.RemindActivity;
import com.example.loren.minesample.VectorActivity;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by loren on 2017/3/7.
 */

public class HomeFragment extends Fragment implements View.OnClickListener {
    @BindView(R.id.fs_tv)
    TextView fsTv;
    @BindView(R.id.flag_tv)
    TextView flagTv;
    @BindView(R.id.move_view)
    TextView moveView;
    @BindView(R.id.remind_tv)
    TextView remindTv;
    @BindView(R.id.wx_vedio)
    TextView wxVedio;
    @BindView(R.id.chat)
    TextView chat;
    @BindView(R.id.open)
    TextView open;
    @BindView(R.id.vector)
    TextView vector;
    @BindView(R.id.container)
    LinearLayout container;
    private Context mContext;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.home_fragment, container, false);
        ButterKnife.bind(this, view);
        return view;

    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mContext = getActivity();
        for (int i = 0; i < container.getChildCount(); i++) {
            if (i % 2 == 0) {
                container.getChildAt(i).setBackgroundColor(ContextCompat.getColor(mContext, R.color.main));
            } else {
                container.getChildAt(i).setBackgroundColor(ContextCompat.getColor(mContext, R.color.match));
            }
        }
        moveView.setOnClickListener(this);
        fsTv.setOnClickListener(this);
        flagTv.setOnClickListener(this);
        remindTv.setOnClickListener(this);
        wxVedio.setOnClickListener(this);
        chat.setOnClickListener(this);
        open.setOnClickListener(this);
        vector.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.move_view:
                startActivity(new Intent(mContext, MoveViewActivity.class), ActivityOptions.makeSceneTransitionAnimation(getActivity(), moveView, "share_tv").toBundle());
                break;
            case R.id.flag_tv:
                startActivity(new Intent(mContext, FlagActivity.class));
                break;
            case R.id.fs_tv:
                startActivity(new Intent(mContext, CustomViewActivity.class), ActivityOptions.makeSceneTransitionAnimation(getActivity()).toBundle());
                break;
            case R.id.remind_tv:
                startActivity(new Intent(mContext, RemindActivity.class));
                break;
            case R.id.wx_vedio:
                startActivity(new Intent(mContext, CopyWxPullActivity.class));
                break;
            case R.id.chat:
                startActivity(new Intent(mContext, ChatActivity.class), ActivityOptions.makeSceneTransitionAnimation(getActivity()).toBundle());
                break;
            case R.id.vector:
                startActivity(new Intent(mContext, VectorActivity.class));
                break;
            case R.id.open:
//                startActivity(new Intent(mContext, OpenScreenActivity.class),ActivityOptions.makeSceneTransitionAnimation(this).toBundle());
                startActivity(new Intent(mContext, OpenScreenActivity.class));
                getActivity().overridePendingTransition(0, 0);
                break;

            default:
                break;
        }
    }
}

File: app/src/main/java/com/example/loren/minesample/fragment/HomeFragment.kt
package com.example.loren.minesample.fragment

import android.app.ActivityOptions
import android.content.Intent
import android.support.v4.content.ContextCompat
import android.view.View
import com.example.loren.minesample.*
import com.example.loren.minesample.annotation.AnimationClick
import com.example.loren.minesample.annotation.AnimationClickType
import com.example.loren.minesample.annotation.LorenInject
import com.example.loren.minesample.base.ui.BaseFragment
import kotlinx.android.synthetic.main.home_fragment.*

/**
 * Copyright © 2017/3/7 by loren
 */

class HomeFragment : BaseFragment() {
    override fun initWidgets() {
        LorenInject.into(this)
        for (i in 0 until container!!.childCount) {
            if (i % 2 == 0) {
                container!!.getChildAt(i).setBackgroundColor(ContextCompat.getColor(mContext, R.color.main))
            } else {
                container!!.getChildAt(i).setBackgroundColor(ContextCompat.getColor(mContext, R.color.match))
            }
        }
    }

    @AnimationClick(viewIds = [R.id.move_view, R.id.fs_tv, R.id.flag_tv, R.id.remind_tv, R.id.wx_vedio, R.id.chat, R.id.open, R.id.vector], animType = AnimationClickType.ALPHA)
    override fun onClick(v: View) {
        when (v.id) {
            R.id.move_view -> startActivity(Intent(mContext, MoveViewActivity::class.java), ActivityOptions.makeSceneTransitionAnimation(activity, move_view, "share_tv").toBundle())
            R.id.flag_tv -> startActivity(Intent(mContext, FlagActivity::class.java))
            R.id.fs_tv -> startActivity(Intent(mContext, CustomViewActivity::class.java), ActivityOptions.makeSceneTransitionAnimation(activity).toBundle())
            R.id.remind_tv -> startActivity(Intent(mContext, RemindActivity::class.java))
            R.id.wx_vedio -> startActivity(Intent(mContext, CopyWxPullActivity::class.java))
            R.id.chat -> startActivity(Intent(mContext, ChatActivity::class.java), ActivityOptions.makeSceneTransitionAnimation(activity).toBundle())
            R.id.vector -> startActivity(Intent(mContext, VectorActivity::class.java))
            R.id.open -> {
                //                startActivity(new Intent(mContext, OpenScreenActivity.class),ActivityOptions.makeSceneTransitionAnimation(this).toBundle());
                startActivity(Intent(mContext, OpenScreenActivity::class.java))
                mContext.overridePendingTransition(0, 0)
            }
        }
    }

    override fun useTitleBar() = false

    override fun setListeners() {
    }

    override fun onWidgetsClick(v: View) {

    }

    override fun bindLayout() = R.layout.home_fragment
}
